cmd/goal/asset.go

```diff
@@ -654,15 +663,15 @@ var freezeAssetCmd = &cobra.Command{
 	},
 }
 
-func assetDecimalsFmt(amount uint64, decimals uint32) string {
+func assetDecimalsFmt(amount uint64, decimals uint64) string {
```

Comment:
Decimals is a `uint32` in the transaction, why change it in this helper function?

Reply:
It was `uint32` in v1, but it looks like it was changed to `uint64` in v2: 

https://github.com/algorand/go-algorand/blob/995ae47e80c50e7632034cac8a70b7d6434d03e3/daemon/algod/api/server/v2/generated/types.go#L229

Reply:
Decimals can only be 0-19 in the spec: https://github.com/algorandfoundation/specs/blob/44fa607d6051730f5264526bf3c108d51f0eadb6/dev/ledger.md?plain=1#L494

In the block it's represented as uint32: https://github.com/algorand/go-algorand/blob/995ae47e80c50e7632034cac8a70b7d6434d03e3/data/basics/userBalance.go#L367

The TEAL opcode seems to be uint64, but it doesn't need to be.

Reply:
I don't think anything would break regardless of this change (one way or the other)

Reply:
Ah interesting, thanks for pointing it out. I took another look, and I don't think the decimal formatting would break either, and I thought it would make more sense to be consistent with the asset params API definition.